dmriprep/utils/vectors.py

```diff
@@ -177,6 +179,105 @@ def to_filename(self, filename, filetype='rasb'):
         else:
             raise ValueError('Unknown filetype "%s"' % filetype)
 
+class BVALScheme:
```

Comment:
I would probably go with a single function that accepts a `DiffusionGradientTable` object for this.

The function would return only a list of nonzero shell centers and the number of shells would be just the `len()` of that (which can be easily calculated by the user).

WDYT?